lib/storage-service.js

```diff
@@ -111,8 +111,17 @@ class StorageService {
    */
   _getLeaseManager(token, secret, ProviderType) {
     const secretID = `/${ProviderType.prototype.constructor.name}/${token}/${secret}`;
+    let tokenManager;
+
+    if (typeof ProviderType.prototype.requireVaultToken === 'function') {
```

Comment:
I expect `requireVaultToken` to be called if we're checking to see that it's a function. Maybe define it as a boolean property on the provider instead?